### PR TITLE
[fix][ci] Fix OWASP Dependency Check download by using NVD API key

### PR DESCRIPTION
**Pull request by shoothzj**
See more in https://github.com/apache/pulsar/pull/22999. Thanks for @lhotari 
INFRA-TICKET https://issues.apache.org/jira/projects/INFRA/issues/INFRA-26000

### CODE REVIEW

**Comment by shoothzj** [comment]
@lhotari It seems download is still failed, do we need to merge this first?

File: pom.xml [hunk 1]
```diff
@@ -192,7 +192,7 @@
     <maven-checkstyle-plugin.version>3.3.1</maven-checkstyle-plugin.version>
     <maven-compiler-plugin.version>3.12.1</maven-compiler-plugin.version>
     <maven-surefire-plugin.version>3.2.5</maven-surefire-plugin.version>
-    <dependency-check-maven.version>9.2.0</dependency-check-maven.version>
+    <dependency-check-maven.version>10.0.1</dependency-check-maven.version>
```
Comment (lhotari) [inline pom.xml:195]: Should use at least 10.0.2 version

**Review by lhotari — CHANGES_REQUESTED** [review]

**Comment by lhotari** [comment]
> @lhotari It seems download is still failed, do we need to merge this first?

Yes. Another option would be to create the PR branch to apache/bookkeeper so that it has access to the secret. The dependency check version must be at least 10.0.2 (there's another Pulsar PR explaining that).

**Comment by shoothzj** [comment]
@lhotari Thanks, I see your description in https://github.com/apache/pulsar/pull/23012.

I suggest we can merge this pr first.

> There's a [mandatory upgrade notice](https://github.com/jeremylong/DependencyCheck?tab=readme-ov-file#mandatory-upgrade-notice) about upgrading to 10.0.2 version.

**Comment by lhotari** [comment]
> I suggest we can merge this pr first.

That works for me, but why not just upgrade to 10.0.2 in this PR?

**Review by lhotari — APPROVED** [review]

**Comment by lhotari** [comment]
Looks like 10.0.2 is used in this PR, LGTM.

**Review by lhotari — APPROVED** [review]

**Comment by lhotari** [comment]
There's a error building the docker image
```
[INFO] Package netcat is a virtual package provided by:
[INFO]   netcat-traditional 1.10-48
[INFO]   netcat-openbsd 1.226-1ubuntu2
[INFO] 
[INFO] E: Package 'netcat' has no installation candidate
```
I believe `netcat-openbsd` should be used.

**Review by lhotari — APPROVED** [review]

**Pull request merged**